### ISSUE TITLE
Rule proposal: `no-anonymous-default-export`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#2264]
### Description

Implement a replacement rule for existing rule [import/no-anonymous-default-export](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-anonymous-default-export.md).

### Fail

```js
export default function () {}
```

```js
export default class {}
```

```js
export default () => {}
```

### Pass

```js
class A {}
export default A;
```

```js
export default class A {};
```

```js
function a {}
export default a;
```

```js
export default function a {}
```

```js
export default [];
export default {};
export default 1;
export default whatever;
```

### Additional Info

For now, I only care about functions and classes, but we can check others in the future.

Maybe we can also suggest a fix based on the filename.

**Comment by fregante** [comment]
Extracted from https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1089#issuecomment-1899598314
